projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -0,0 +1,207 @@
+})
+export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
+  @HostBinding('class.lg-sort-code-field') class = true;
+
+  private uniqueId = nextUniqueId++;
+  sortCodeFormGroup: FormGroup;
+  first: FormControl;
+  second: FormControl;
+  third: FormControl;
+  valueChanges: Subscription;
+  hasError: boolean;
+
+  @Input() value: string;
+  @Input() disabled: false;
+  @Input() labelId = `lg-input-sort-code-label-${this.uniqueId}`;
+  @Input() firstId = `lg-input-sort-code-first-${this.uniqueId}`;
+  @Input() secondId = `lg-input-sort-code-second-${this.uniqueId}`;
+  @Input() thirdId = `lg-input-sort-code-third-${this.uniqueId}`;
+  @Input() ariaDescribedBy: string;
+
+  private _hintElement: LgHintComponent;
+  @ContentChild(LgHintComponent)
+  set hintElement(element: LgHintComponent) {
+    this.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.ariaDescribedBy,
+      this._hintElement,
+      element,
+    );
+    this._hintElement = element;
+  }
+
+  private _validationElement: LgValidationComponent;
+  @ContentChild(LgValidationComponent)
+  set errorContentElement(element: LgValidationComponent) {
+    this.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.ariaDescribedBy,
+      this._validationElement,
+      element,
+    );
+    this._validationElement = element;
+  }
+
+  @ViewChild('sortCodeFormDirective')
+  formGroupDirective: FormGroupDirective;
+
+  constructor(
+    private domService: LgDomService,
+    private errorState: LgErrorStateMatcher,
+    @Self()
+    @Optional()
+    private ngControl: NgControl,
+    private parentFormGroupDirective: FormGroupDirective,
+  ) {
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+    this.first = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.second = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.third = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.sortCodeFormGroup = new FormGroup(
+      {
+        first: this.first,
+        second: this.second,
+        third: this.third,
+      },
+      {
+        updateOn: 'blur',
+      },
+    );
+  }
+
+  ngOnInit(): void {
+    // append internal validators to external validators if applicable.
+    if (this.ngControl && this.ngControl.control) {
+      this.ngControl.control.setValidators([this.validate.bind(this)]);
+      this.ngControl.control.updateValueAndValidity();
+    }
+
+    this.valueChanges = this.sortCodeFormGroup.valueChanges.subscribe(
+      (sortCode: SortCodeField) => {
+        const first = sortCode.first || '';
+        const second = sortCode.second || '';
+        const third = sortCode.third || '';
+        this.onChange(`${first}${second}${third}`);
+      },
+    );
+
+    // submit the group when the parent form is submitted
```

Comment:
this could do with some test coverage around it

Reply:
I created a PR into this branch @owensgit  (https://github.com/owensgit/canopy/pull/10)

Reply:
i pushed this straight into the branch as Owen is unable to review this for a few days